mne/io/nirx/nirx.py

```diff
@@ -21,15 +21,23 @@
 
 
 @fill_doc
-def read_raw_nirx(fname, preload=False, verbose=None):
+def read_raw_nirx(fname, saturated='ignore', preload=False, verbose=None):
     """Reader for a NIRX fNIRS recording.
 
-    This function has only been tested with NIRScout devices.
-
     Parameters
     ----------
     fname : str
         Path to the NIRX data folder or header file.
+    saturated : str
+        Replace saturated segments of data with NaNs.
+        If "ignore" (default) the measured data is returned, even if it
+        contains measurements while the amplifier was saturated.
+        If "nan" the returned data will contain NaNs during time segments
+        when the amplifier was saturated.
+        If "annotate" the returned data will contain annotations specifying
+        sections the saturate segments and the data will contain NaNs.
```

Comment:
This seems unexpected to me. I would have expected that the purpose of this option would be to return real data (no NaNs) so you can filter etc without nan issues. And could then exclude the bad segments.

Or I guess you could then use the annotations to fill in the bad segments using interpolation between the end points of good data.

@rderollepot sorry if this was discussed already, but is this the intended behaviour?

Reply:
Not returning NaNs here will be a big rewrite though :(

Reply:
I agree it might be nicer in `annotate` mode to give the actual values but annotate with `BAD_SATURATED`, and this makes sense as the default behavior. I think it's worth rewriting for example to pass a list of the start/stop intervals that would need to be added as annotations.

Then a test could be that `annotate` mode should give the same results as `nan` mode + `annotate_nan`, other than the `description` of the annotations being different (the former being `BAD_SATURATED` and the latter being `BAD_NAN` or so)

Reply:
> I agree it might be nicer in annotate mode to give the actual values but annotate with BAD_SATURATED, and this makes sense as the default behavior.

Agreed, this seems a logical default behaviour. However, I don’t have time to do this now. Getting the PR fixed, adding tests, and getting CIs green already took more time than anticipated. This is now in a good state for anyone to take over and change default behaviour.

Reply:
> This seems unexpected to me. I would have expected that the purpose of this option would be to return real data (no NaNs) so you can filter etc without nan issues. And could then exclude the bad segments.
> 
> Or I guess you could then use the annotations to fill in the bad segments using interpolation between the end points of good data.
> 
> @rderollepot sorry if this was discussed already, but is this the intended behaviour?

No I think you are right, after some digging into the original PR, it seems that this topic had been quite heavily discussed, and the conclusion that cbrnr drawn at that time can be found [here](https://github.com/mne-tools/mne-python/pull/7936#discussion_r449058046).

I also agree that 'annotate' (without NaNs) would make a better default behaviour than 'ignore', as 'ignore' now seems to me a quite dangerous option that one should carefully choose.